retrieval/target.go

```diff
@@ -196,11 +198,14 @@ func (t *Target) Update(cfg *config.ScrapeConfig, baseLabels, metaLabels clientm
 	if cfg.BasicAuth != nil {
 		t.url.User = url.UserPassword(cfg.BasicAuth.Username, cfg.BasicAuth.Password)
 	}
+	t.url.RawQuery = cfg.Params.Encode()
```

Comment:
The path can already be relabeled.
Query params seems like a very advanced use case (example appreciated). Something like `__query_<name>__` which contains a joined list of the values (like for Consul tags).